File: Java/src/_0035_Search_Insert_Position/Solution.java
package _0035_Search_Insert_Position;

/**
 * Date: 2020/9/18 15:15
 * Content: 二分查找
 */
public class Solution {

    public int searchInsert(int[] nums, int target) {

        int l = 0, r = nums.length - 1;
        while (l <= r) {
            int mid = (r - l) / 2 + l;
            if (target == nums[mid]) return mid;

            if (target > nums[mid]) l = mid + 1;
            else r = mid - 1;
        }

        return l;
    }

    public static void main(String[] args) {

        int[] nums = {1, 3, 5, 6};
        int target = 0;
        System.out.println((new Solution()).searchInsert(nums, target));
    }
}

File: Python/_0035_Search_Insert_Position/Solution.py
# coding=utf-8
# Date: 2021/5/22 10:01
from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l = 0; r = len(nums)
        while l < r:
            mid = l + r >> 1
            if nums[mid] >= target:
                r = mid
            else:
                l = mid + 1
        return r


if __name__ == "__main__":
    nums = [1, 3, 5, 6]
    print(Solution().searchInsert(nums, 5))  # 2
